apps/api/src/domain/models/user-management/contributor/commands/create-contributor/create-contributor.command.ts

```diff
@@ -0,0 +1,36 @@
+import { AggregateCompositeIdentifier, ICommandBase } from '@coscrad/api-interfaces';
+import { Command } from '@coscrad/commands';
+import { NestedDataType, NonEmptyString } from '@coscrad/data-types';
+import { CoscradDate } from '../../../utilities';
+import { ContributorCompositeIdentifier } from '../contributor-composite-identifier';
+
+@Command({
+    type: 'CREATE_CONTRIBUTOR',
+    label: 'Create Contributor',
+    description: 'Create a new COSCRAD contributor',
+})
+export class CreateContributor implements ICommandBase {
+    @NestedDataType(ContributorCompositeIdentifier, {
+        label: 'Composite Identifier',
+        description: 'system-wide unique identifier',
+    })
+    readonly aggregateCompositeIdentifier: AggregateCompositeIdentifier;
+
+    @NonEmptyString({
+        label: "the contributor's first name",
+        description: "The Contributor's first name",
+    })
+    readonly firstName: string;
+
+    @NonEmptyString({
+        label: "the contributor's last name",
+        description: "The Contributor's last name",
+    })
+    readonly lastName: string;
+
+    @NestedDataType(CoscradDate, {
+        label: 'date of birth',
+        description: "The Contributor's Date of Birth",
+    })
+    readonly dateOfBirth?: CoscradDate;
```

Comment:
This won't work. We don't have a date picker form element. We probably want to accept a `MM-DD-YYYY` format and then translate this in the handler (and on the event). What I recommend for now is that we comment out the dateOfBirth property and leave a TODO to support it later via an additional command `ADD_DATE_OF_BIRTH_FOR_CONTRIBTOR`.

Reply:
So we have decided to make this a non-empty string. As a matter of UX, we should let the front-end know that this is actually a date. Let’s add a story for this.

Reply:
[Here](https://www.pivotaltracker.com/story/show/187391138) is the story.